test/common/http/http1/conn_pool_test.cc

```diff
@@ -1125,6 +1150,99 @@ TEST_F(Http1ConnPoolImplTest, PendingRequestIsConsideredActive) {
   EXPECT_EQ(1U, cluster_->stats_.upstream_cx_destroy_local_.value());
 }
 
+// Schedulable callback that can track it's destruction.
+class MockDestructSchedulableCallback : public Event::MockSchedulableCallback {
+public:
+  MockDestructSchedulableCallback(Event::MockDispatcher* dispatcher)
+      : Event::MockSchedulableCallback(dispatcher) {}
+  MOCK_METHOD0(Die, void());
+
+  ~MockDestructSchedulableCallback() override { Die(); }
+};
+
+// Connection pool for test that holds on to a schedulable callback that has destructor detection.
+class ConnPoolImplNoDestructForTest : public ConnPoolImplForTest {
+public:
+  ConnPoolImplNoDestructForTest(Event::MockDispatcher& dispatcher,
+                                Upstream::ClusterInfoConstSharedPtr cluster,
+                                Random::RandomGenerator& random_generator,
+                                MockDestructSchedulableCallback* upstream_ready_cb)
+      : ConnPoolImplForTest(dispatcher, cluster, random_generator, upstream_ready_cb) {}
+
+  ~ConnPoolImplNoDestructForTest() override {} = default;
```

Comment:
Should be:
~ConnPoolImplNoDestructForTest() override = default;

Reply:
I ended up getting rid of it, I dealt with the client destruction in the test instead of in this special connpool